### PR TITLE
In-graph async replica training for MNIST

### PR DESCRIPTION
**Pull request by windreamer**
@mckelvin pls review

### CODE REVIEW

File: examples/mnist/mnist.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    },
+    {
+        "name": "worker",
+        "num": 5
+    },
+]
+
+_lock = RLock()
+mnist = read_data_sets("MNIST_data/", one_hot=True)
+with cluster(jobs_def, master='cu1', quiet=False) as targets:
```
Comment (mckelvin) [inline examples/mnist/mnist.py]: It should be better to pass the mesos master address from something like `argv[1]`.

**Comment by bhack** [comment]
Can be used zk:// addresses here? So we don't need to manually resolve the leader every time.

**Comment by windreamer** [comment]
@bhack sure we can use `zk://` address for master, it is supported by `pymesos`

**Comment by mckelvin** [comment]
LGTM.

**Comment by bhack** [comment]
We can test this on GPU. Do you plan to merge?

**Pull request merged**